libs/cluster/Server/Replication/ReplicaOps/ReplicationReplicaAofSync.cs

```diff
@@ -48,14 +59,23 @@ public unsafe void ProcessPrimaryStream(byte* record, int recordLength, long pre
                             )
                         {
                             logger?.LogWarning("MainMemoryReplication: Skipping from {ReplicaReplicationOffset} to {currentAddress}", ReplicationOffset, currentAddress);
-                            storeWrapper.appendOnlyFile.Initialize(currentAddress, currentAddress);
+                            storeWrapper.appendOnlyFile.SafeInitialize(currentAddress, currentAddress);
                             ReplicationOffset = currentAddress;
                         }
                     }
                 }
 
-                // Address check
-                if (ReplicationOffset != storeWrapper.appendOnlyFile.TailAddress)
+                var tail = storeWrapper.appendOnlyFile.TailAddress;
+                var pageBits = storeWrapper.appendOnlyFile.UnsafeGetLogPageSizeBits();
```

Comment:
can this be done once instead of for every single call?

Reply:
Done here: https://github.com/microsoft/garnet/pull/783/commits/26d30023c801121346eb738042fa3f507d66a264